src/engine/GroupBy.h

```diff
@@ -14,26 +14,22 @@
 #include <vector>
 
 #include "engine/GroupByHashMapOptimization.h"
+#include "engine/Join.h"
 #include "engine/Operation.h"
 #include "engine/QueryExecutionTree.h"
-#include "engine/sparqlExpressions/RelationalExpressionHelpers.h"
 #include "engine/sparqlExpressions/SparqlExpressionPimpl.h"
 #include "engine/sparqlExpressions/SparqlExpressionValueGetters.h"
 #include "parser/Alias.h"
-#include "parser/ParsedQuery.h"
 #include "util/TypeIdentity.h"
 
-using std::string;
-using std::vector;
-
-// Forward declarations for internal member function
-class IndexScan;
-class Join;
```

Comment:
I deliberately used a forward declaration here to reduce the number of required headers.
What is your rationale?

Reply:
`IndexScan` is not required in the header and in case `Join` is ever renamed including the header will properly let the code fail on compilation rather than giving errors when linking

Reply:
Also I just checked, there are only 2 Files (both cpp) currently including GroupBy, soon to be 3 with the upcoming PR that the only new unique header for `GroupBy.cpp` is likely to be `util/JoinAlgorithms/JoinAlgorithms.h` all the others are likely already included anyways